src/test/java/picard/vcf/SamTestUtils.java

```diff
@@ -4,30 +4,46 @@
 import htsjdk.samtools.SAMFileWriterFactory;
 import htsjdk.samtools.SamReader;
 import htsjdk.samtools.SamReaderFactory;
+import picard.PicardException;
 
 import java.io.File;
 import java.io.IOException;
 
 public class SamTestUtils {
     /**
-     * Useful test method.  Creates a (temporary) indexed BAM so that we can store a sam in the testdata set.
+     * Useful test method.  Creates a (temporary) indexed BAM or CRAM so that we can store a sam in the testdata set.
      *
      * @param samFile the sam file to convert to bam and index
-     * @return File a (temporary) bam file (index file is created in same path).
+     * @param type the type of file to be created
+     * @param referenceFasta reference to be used if creating CRAM
+     * @return File a (temporary) bam or cram file (index file is created in same path).
      */
-    public static File createIndexedBam(final File samFile, final File tempFilePrefix) throws IOException {
-        final File output = File.createTempFile(tempFilePrefix.getAbsolutePath(), ".bam");
+
+    public static File createIndexedBamOrCram(final File samFile, final File tempFilePrefix, final SamReader.Type type, final File referenceFasta) throws IOException {
+        if (type != SamReader.Type.BAM_TYPE && type != SamReader.Type.CRAM_TYPE) {
+            throw new PicardException("Cannot create indexed file of type " + type);
+        }
+
+        if (type == SamReader.Type.CRAM_TYPE && referenceFasta == null) {
+            throw new PicardException("Cannot create indexed CRAM file without reference");
+        }
+
+        final File output = File.createTempFile(tempFilePrefix.getAbsolutePath(), "." + type.fileExtension());
         output.deleteOnExit();
-        final File indexFile = new File(output.getAbsolutePath() + ".bai");
+        final File indexFile = new File(output.getAbsolutePath() + "." + type.indexExtension());
```

Comment:
One subtle thing going on here is that this is manually creating a ".crai" file for the index, but the writer defies intuition and actually creates a ".bai" CRAM, since thats what the htsjdk implementation prefers. So the deleteOnExit() below won't delete the .bai file, and its the .crai file that actually will be used.

Reply:
Are there any tests that compare the results from running on a cram with the results of running on the equivalent bam? I'd strongly suggest adding tests that do that if possible.

Reply:
@cmnbroad thanks for catching that!  In double checking this it looks like I wasn't treating bams correctly either, my code was assuming a .bam.bai index, but it looks like the writer creates a simple .bai index for a bam and a .cram.bai index for a cram. Should be all fixed now.

Reply:
Also added tests to compare crosscheck on a bam to crosscheck on the equivalent cram, good suggestion.